contrib/drivers/drivers.go

```diff
@@ -129,3 +129,28 @@ func Clean(d VolumeDriver) VolumeDriver {
 
 	return d
 }
+
+type MetricDriver interface {
+	//Any initialization the metric driver does while starting.
+	Setup() error
+	//Any operation the metric driver does while stopping.
+	Teardown() error
+
+	CollectMetrics(metricList []string, instanceID string) ([]model.MetricSpec, error)
```

Comment:
I suggest you can return []*model.MetricSpec here.

Reply:
It is a pointer to struct array, so made it as *[]model.MetricSpec

Reply:
No, I think it's better to define an array of pointers to struct.